pynsot/commands/cmd_circuits.py

```diff
@@ -0,0 +1,378 @@
+    callback=callbacks.transform_attributes,
+)
+@click.option(
+    '-A',
+    '--endpoint-a',
+    metavar='INTERFACE_ID',
+    type=int,
+    help='Unique ID of the interface of the A side of the Circuit',
+)
+@click.option(
+    '-n',
+    '--name',
+    metavar='NAME',
+    type=str,
+    help='The name of the Circuit.',
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    type=int,
+    help='Unique ID of the Site this Circuit is under. [required]',
+    callback=callbacks.process_site_id,
+)
+@click.option(
+    '-Z',
+    '--endpoint-z',
+    metavar='INTERFACE_ID',
+    type=int,
+    help='Unique ID of the interface on the Z side of the Circuit',
+)
+@click.pass_context
+def add(ctx, attributes, endpoint_a, name, site_id, endpoint_z):
+    """
+    Add a new Circuit.
+
+    You must provide the A side of the circuit using the -A/--endpoint-a
+    option. The Z side is recommended but may be left blank, such as in cases
+    where it is not an Interface that is tracked by NSoT (like a provider's
+    interface).
+
+    The name (-n/--name) is optional. If it is not specified, it will be
+    generated for you in the form of:
+    {device_a}:{interface_a}_{device_z}:{interface_z}
+
+    If you wish to add attributes, you may specify the -a/--attributes
+    option once for each key/value pair.
+    """
+
+    data = ctx.params
+
+    if endpoint_a is None:
+        raise click.UsageError('Missing option "-A" / "--endpoint-a"')
+
+    # Remove empty values to facilitate default assignment
+    if name is None:
+        data.pop('name')
+    if endpoint_z is None:
+        data.pop('endpoint_z')
+
+    ctx.obj.add(data)
+
+
+# List
+@cli.group(invoke_without_command=True)
+@click.option(
+    '-a',
+    '--attributes',
+    metavar='ATTRS',
+    help='Filter Circuits by matching attributes (format: key=value).',
+    multiple=True,
+)
+@click.option(
+    '-g',
+    '--grep',
+    is_flag=True,
+    help='Display list results in a grep-friendly format.',
+    default=False,
+    show_default=True,
+)
+@click.option(
+    '-i',
+    '--id',
+    metavar='ID',
+    help='Unique ID of the Circuit being retrieved.',
+)
+@click.option(
+    '-l',
+    '--limit',
+    metavar='LIMIT',
+    help='Limit result to N resources.',
+)
+@click.option(
+    '-N',
+    '--natural-key',
+    is_flag=True,
+    help='Display list results by their natural key',
+    default=False,
+    show_default=True,
+)
+@click.option(
+    '-o',
+    '--offset',
+    metavar='OFFSET',
+    help='Skip the first N resources.',
+)
+@click.option(
+    '-q',
+    '--query',
+    metavar='QUERY',
+    help='Perform a set query using Attributes and output matching Circuits.'
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    help='Unique ID of the Site this Circuit is under.',
+    callback=callbacks.process_site_id,
+)
+@click.pass_context
+def list(ctx, attributes, grep, id, limit, natural_key, offset, query,
```

Comment:
I feel like we should have CLI filter options for anything we can filter on using the API. Since we've yet to update the docs to include circuits, you can reference https://github.com/dropbox/nsot/blob/master/nsot/api/filters.py#L153 (we're missing name, endpoint_a, endpoint_z).

I see that you substitued `-i` for a single object lookup, so maybe `name` isn't relevant here. But I'm also wondering if some people might find that confusing.

See how `networks list` results can be filtered for a reference of the filter options I'm talking about.